src/session_impl.cpp

```diff
@@ -4829,6 +4829,14 @@ namespace {
 			return std::make_pair(ptr_t(), false);
 		}
 
+		if (params.ti
+			&& params.info_hash != sha1_hash(0)
```

Comment:
what do you think about using `is_all_zeros` instead?

Reply:
probably a good idea